src/PhpDocParser/PhpStan.php

```diff
@@ -0,0 +1,159 @@
+<?php declare(strict_types = 1);
+
+namespace RstGroup\ObjectBuilder\PhpDocParser;
+
+use PhpParser\Node\Stmt;
+use PhpParser\Parser;
+use PHPStan\PhpDocParser\Lexer\Lexer;
+use PHPStan\PhpDocParser\Parser\PhpDocParser as PhpStanPhpDocParser;
+use PHPStan\PhpDocParser\Parser\TokenIterator;
+use ReflectionClass;
+use ReflectionParameter;
+use RstGroup\ObjectBuilder\BuildingError;
+use RstGroup\ObjectBuilder\PhpDocParser;
+
+final class PhpStan implements PhpDocParser
+{
+    /** @var PhpStanPhpDocParser */
+    private $phpDocParser;
+    /** @var Parser */
+    private $phpParser;
+
+    /** @codeCoverageIgnore */
+    public function __construct(
+        PhpStanPhpDocParser $phpDocParser,
+        Parser $phpParser
+    ) {
+        $this->phpDocParser = $phpDocParser;
+        $this->phpParser = $phpParser;
+    }
+
+    public function isListOfObject(string $comment, string $parameterName): bool
+    {
+        $node = $this->phpDocParser->parse(new TokenIterator((new Lexer())->tokenize($comment)));
+
+        foreach ($node->getParamTagValues() as $node) {
+            if ('$' . $parameterName === $node->parameterName) {
+                $typeName = $node->type->type->name;
+                if ($this->isScalar($typeName)) {
+                    continue;
+                }
+
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public function getListType(string $comment, ReflectionParameter $parameter): string
+    {
+        $node = $this->phpDocParser->parse(new TokenIterator((new Lexer())->tokenize($comment)));
+
+        foreach ($node->getParamTagValues() as $node) {
+            if ($node->parameterName === '$' . $parameter->getName()) {
+                $type = $node->type->type;
+
+                /** @var ReflectionClass $class */
+                $class = $parameter->getDeclaringClass();
+                /** @var string $fileName */
+                $fileName = $class->getFileName();
+                /** @var string $phpFileContent */
+                $phpFileContent = file_get_contents($fileName);
+                /** @var Stmt[] $parsedFile */
+                $parsedFile = $this->phpParser->parse($phpFileContent);
+
+                $namespace = $this->getNamespaceStmt($parsedFile);
+                $uses = $this->getUseStmts($namespace);
+                $namespaces = $this->getUsesNamespaces($uses);
+
+                return $this->getFullClassName($type->name, $namespaces, $class);
+            }
+        }
+
+        throw new BuildingError();
+    }
+
+    private function isScalar(string $value): bool
+    {
+        $scalars = [
+            'string',
+            'bool',
+            'int',
+            'float',
+            'double',
+            'mixed',
```

Comment:
Mixed === scalar?

Reply:
In context of this private function, it is. But I just didn't find better name so I'm open for alternatives name.